addon/components/frost-pick-list.js

```diff
@@ -1,6 +1,125 @@
 import Ember from 'ember'
-const {Component} = Ember
+const {A, Component, copy} = Ember
+import computed, {readOnly} from 'ember-computed-decorators'
+import {sort} from 'ember-frost-sort'
+import layout from '../templates/components/frost-pick-list'
+import {PropTypes} from 'ember-prop-types'
 
 export default Component.extend({
-  classNames: ['frost-pick-list']
+  // == Dependencies ==========================================================
+
+  // == Properties ============================================================
+
+  classNames: ['frost-pick-list'],
+  itemsCached: A([]),
+  inputFilter: '',
+  onlySelected: false,
+  layout,
+
+  // == PropTypes =============================================================
+
+  propTypes: {
+    // options
+    listComponent: PropTypes.object.isRequired,
+    selectedItems: PropTypes.arrayOf(PropTypes.object),
+    items: PropTypes.arrayOf(PropTypes.object),
+    sortOrder: PropTypes.arrayOf(PropTypes.string),
+    sortingProperties: PropTypes.arrayOf(PropTypes.object),
+    sortOrderMax: PropTypes.number
+  },
+
+  getDefaultProps () {
+    return {
+      // options
+      items: A([]),
+      sortOrder: A([]),
+      selectedItems: A([]),
+      sortingProperties: A([]),
+      sortOrderMax: 1
+    }
+  },
+
+  // == Computed Properties ===================================================
+
+  @readOnly
+  @computed('onlySelected')
+  /**
+   * show all record or just selected
+   * @param {Boolean} onlySelected - should show only selected?
+   * @returns {Boolean} should show only selected?
+   */
+  onlySelectedComputed (onlySelected) {
```

Comment:
can we rename this cp? something without `computed` :)

Reply:
Don't you think post fixing `computed` helps differentiate between variables that are computed or not in the template?

Reply:
@mo0om  What value does it add in the template?  Why do I care how a value is populated - I just want to use it.  It's the same as whether I care it comes from a service, store, Ajax call, was hard-coded, etc - it's all irrelevant.  If I know that it's a computed property versus it's not what is my decision tree for its usage in the template?  Will I do something different if I know this information?

Reply:
I agree with you that it's not making a difference in the decision tree but I personally like to know what kind of variable I have to deal with when I look at the template. Especially when we deal with components inside components. There is also a special relationship between a computed property and its template in that it must be present in the template for it to work. I think naming it with a `computed` post fix reinforce this.

Reply:
> I agree with you that it's not making a difference in the decision tree
then it should not be there.  We do not follow Hungarian notation (https://en.wikipedia.org/wiki/Hungarian_notation)

> Especially when we deal with components inside components
the level of nested usage does not begin to add value the more levels there are, as my usage does not change "further in" once I know it's a computed property

> There is also a special relationship between a computed property and its template in that it must be present in the template for it to work.
Yes, a computed property must first be accessed before its value is computed.  And doing so in a template will trigger this computation.  But you should be accessing a property, computed or otherwise, in a template because you want to display (or interact with) its value and not because you need to trigger the computation of a computed property's value because other code is dependent on.  That is a horrible pattern.  I'm sure that is not what you are doing, but that is what your statement alludes to.  

Computed properties do not only have to be accessed via a template and can be by code within the object itself as well.  Are you suggesting that all computed properties be postfixed with `Computed` (though again we do not follow Hungarian notation)?  Or only ones being used in templates?  What happens when you want to use one in a template later that you weren't previously - do you have to rename it?  There is no value added by this use, and the result of the actions taken by the questions I just posed we're actually *losing* value.  To this point of inconsistency and lost value, I notice that you did not postfix `subTitle` with `Computed` though its used in the _addon/templates/components/frost-pick-list.hbs_ template ;-)

Reply:
lol, true